compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1565,6 +1549,21 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
         val selType = rawSelectorTpe match
           case c: ConstantType if tree.isInline => c
           case otherTpe => otherTpe.widen
+        /** Extractor for match types hidden behind an AppliedType/MatchAlias */
```

Comment:
Reverts changes from https://github.com/lampepfl/dotty/pull/14639/files#diff-8c9ece1772bd78160fc1c31e988664586c9df566a1d22ff99ef99dd6d5627a90R1518

Match Type completion is also fixed by this commit without using this extractor.